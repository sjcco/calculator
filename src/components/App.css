.App {
  text-align: center;
}

.skin {
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
  flex-direction: column;
  width: 100%;
}

.nav-wrapper {
  background-color: black;
}

.nav-wrapper a {
  text-decoration: none;
  color: white;
}

.container {
  display: flex;
  justify-content: space-around;
}

.brand {
  text-align: center;
  font-weight: bold;
  line-height: 63px;
}

.nav-links {
  display: flex;
  list-style: none;
}

.nav-links li {
  margin-right: 1.5rem;
  padding: 5px 0;
}

.nav-links a {
  padding: 0 0.5rem;
}

.nav-links li:hover {
  text-decoration: underline;
}

.active {
  background-color: #0089a6;
  border-radius: 16px;
}

.content-container {
  width: 85%;
  margin: 0 auto;
}

.calc-container {
  margin-top: 2rem;
  display: flex;
  justify-content: space-between;
}

.calculator {
  max-width: 700px;
}

.button-panel {
  background-color: gray;
}

.display {
  background-color: gray;
  height: 100px;
  flex-direction: row;
  justify-content: space-between;
  font-weight: bold;
}

.display p {
  padding: 0 1.5rem 0 1.5rem;
}

.btn-group {
  width: 100%;
  height: 100px;
  display: flex;
  justify-content: center;
}

.button {
  width: 25%;
  height: 100%;
  padding: 1.5rem 0;
  border: 1px solid gray;
}

.zero {
  width: 50%;
}

.bg-gray {
  background-color: rgb(230, 230, 230);
}

.bg-orange {
  background-color: orange;
}
