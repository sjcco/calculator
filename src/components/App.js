import React from 'react';
import './App.css';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
// eslint-disable-next-line no-unused-vars
import calculate from '../logic/calculate';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      // eslint-disable-next-line
      operation: null,
    };

    this.handleClick = btnName => {
      this.setState(prevState => calculate(prevState, btnName));
    };
  }

  render() {
    const { next, total } = this.state;
    return (
      <>
        <Display result={next || total || '0'} />
        <ButtonPanel clickHandler={this.handleClick} />
      </>
    );
  }
}

export default App;
