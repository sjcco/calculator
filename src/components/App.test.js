import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

test('render page correctly', () => {
  const component = renderer.create(<App />).toJSON();
  expect(component).toMatchSnapshot();
});
