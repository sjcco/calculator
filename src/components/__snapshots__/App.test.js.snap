// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render page correctly 1`] = `
Array [
  <nav
    className="nav-wrapper"
  >
    <div
      className="container"
    >
      <a
        className="brand"
        href="/"
      >
        Math-Magicians
      </a>
      <ul
        className="nav-links"
      >
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/quotes"
            onClick={[Function]}
          >
            Quote of the day
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <div
    className="content-container"
  >
    <h1>
      Welcome to our page!
    </h1>
    <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Aliquid itaque eum, non quis sit et dicta vitae quae asperiores expedita, error quos suscipit.
    </p>
    <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Aliquid itaque eum, non quis sit et dicta vitae quae asperiores expedita, error quos suscipit.
    </p>
  </div>,
]
`;
